beacon_node/eth1/src/http.rs

```diff
@@ -141,22 +180,24 @@ pub async fn get_block(
     ]);
 
     let response_body = send_rpc_request(endpoint, "eth_getBlockByNumber", params, timeout).await?;
-    let hash = hex_to_bytes(
-        response_result(&response_body)?
+    let hash: Vec<u8> = hex_to_bytes(
+        response_result_or_error(&response_body)
+            .ok()
```

Comment:
When we use `ok()` here we're still flattening all the different error types into "No result field was returned for block".

So, if there's an `RpcError::Error(s)` we're still presenting it to the user as if there was no result field.